core/src/net/network_thread.rs

```diff
@@ -339,6 +342,7 @@ impl NetworkThread {
     fn read_tcp(&mut self, event: &EventWithRetries) -> () {
         if let Some(channel_rc) = self.get_channel_by_token(&event.token) {
             let mut channel = channel_rc.borrow_mut();
+            let active = channel.connected();
             loop {
                 match channel.read_frame(&self.buffer_pool) {
```

Comment:
If we managed to read some data i.e. the case `Ok(Some(Frame::Data(data)))`, shouldn't we call `update_lru()`? I'm thinking that we don't want to evict a channel we have read recently from.

(sorry could not put the comment on that match handler as there were no changes there)